controllers/etcd_custodian_controller.go

```diff
@@ -0,0 +1,163 @@
+			return nil, fmt.Errorf("%v/%v etcd is marked for deletion", etcd.Namespace, etcd.Name)
+		}
+		if foundEtcd.UID != etcd.UID {
+			return nil, fmt.Errorf("original %v/%v etcd gone: got uid %v, wanted %v", etcd.Namespace, etcd.Name, foundEtcd.UID, etcd.UID)
+		}
+		return foundEtcd, nil
+	})
+
+	selector, err := metav1.LabelSelectorAsSelector(etcd.Spec.Selector)
+	if err != nil {
+		logger.Error(err, "Error converting etcd selector to selector")
+		return ctrl.Result{}, err
+	}
+
+	cm := NewEtcdDruidRefManager(ec.Client, ec.Scheme, etcd, selector, etcdGVK, canAdoptFunc)
+
+	ss, err := cm.FetchStatefulSet(etcd)
+	if err != nil {
+		return ctrl.Result{
+			Requeue: true,
+		}, err
+	}
+
+	// If no statefulsets could be fetched, requeue for reconcilation
+	if len(ss) < 1 {
+		return ctrl.Result{
+			RequeueAfter: time.Duration(5 * time.Second),
+		}, nil
+	}
+
+	if err := ec.updateEtcdStatus(ctx, etcd, ss[0]); err != nil {
+		return ctrl.Result{
+			Requeue: true,
+		}, err
+	}
+
+	return ctrl.Result{
+		Requeue: false,
+	}, nil
+}
+
+func (ec *EtcdCustodian) updateEtcdStatus(ctx context.Context, etcd *druidv1alpha1.Etcd, sts *appsv1.StatefulSet) error {
+	logger.Infof("Reconciling etcd status in Custodian Controller for etcd statefulset status:%s in namespace:%s", etcd.Name, etcd.Namespace)
+	if err := ec.Get(ctx, types.NamespacedName{Name: etcd.Name, Namespace: etcd.Namespace}, etcd); err != nil {
+		logger.Errorf("Error during fetching ETCD resource in ETCD controller: %v", err)
+		return err
+	}
+
+	etcd.Status.Etcd = druidv1alpha1.CrossVersionObjectReference{
+		APIVersion: sts.APIVersion,
+		Kind:       sts.Kind,
+		Name:       sts.Name,
+	}
+	ready := CheckStatefulSet(etcd, sts) == nil
+	conditions := []druidv1alpha1.Condition{}
+	for _, condition := range sts.Status.Conditions {
+		conditions = append(conditions, convertConditionsToEtcd(&condition))
+	}
+	etcd.Status.Conditions = conditions
+
+	// To be changed once we have multiple replicas.
+	etcd.Status.CurrentReplicas = sts.Status.CurrentReplicas
+	etcd.Status.ReadyReplicas = sts.Status.ReadyReplicas
+	etcd.Status.UpdatedReplicas = sts.Status.UpdatedReplicas
+	etcd.Status.Ready = &ready
+
+	logger.Infof("ETCD status updated for statefulset current replicas: %v, ready replicas: %v, updated replicas: %v", sts.Status.CurrentReplicas, sts.Status.ReadyReplicas, sts.Status.UpdatedReplicas)
+	if err := ec.Status().Update(ctx, etcd); err != nil && !errors.IsNotFound(err) {
+		logger.Errorf("Error while updating ETCD status in custodian controller: %v", err)
+		return err
+	}
+	return nil
+}
+
+// SetupWithManager sets up manager with a new controller and ec as the reconcile.Reconciler
+func (ec *EtcdCustodian) SetupWithManager(mgr ctrl.Manager, workers int) error {
+	builder := ctrl.NewControllerManagedBy(mgr).WithOptions(controller.Options{
+		MaxConcurrentReconciles: workers,
+	})
+
+	return builder.For(&appsv1.StatefulSet{}).Owns(&druidv1alpha1.Etcd{}).Complete(ec)
```

Comment:
Like I mentioned in out-of-band discussion, it would have been better to do this using `For` on `Etcd` and `Watches` on `StatefulSet`. But I am OK if you want to do that in a different PR later. @abdasgupta Please create an issue for this so that we do not forget.

Approving for now.